packages/flutter/test/widgets/framework_test.dart

```diff
@@ -467,4 +467,40 @@ void main() {
       <Key>[null, key1, null, key2, null],
     );
   });
+
+  testWidgets('Element diagnostics', (WidgetTester tester) async {
+    GlobalKey key0;
+    await tester.pumpWidget(new Column(
+      key: key0 = new GlobalKey(),
+      children: <Widget>[
+        new Container(),
+        new Container(key: new GlobalKey()),
+        new Container(child: new Container()),
+        new Container(key: new GlobalKey()),
+        new Container(),
+      ],
+    ));
+    final MultiChildRenderObjectElement element = key0.currentContext;
+
+    final String dump =
+        element.toStringDeep().replaceAll(new RegExp(r'#\d+'), '');
```

Comment:
rather than replacing with nothing, i recommend replacing with something like `#000`, so that if a `#0` appears or disappears from the original input, the test fails.